casper/src/main/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala

```diff
@@ -145,16 +201,20 @@ object RegistrySigGen {
     val toSign: Par = ETuple(Seq(GInt(lastNonce), access))
     val sig         = Ed25519.sign(toSign.toByteArray, secKey)
 
+    val keyHash = Blake2b256.hash(pubKey)
+    val uri     = Registry.buildURI(keyHash)
+
```

Comment:
What is the purpose of this uri?

Reply:
this is the uri under which the contract is found by calling the registry lookup